server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/FastVectorHighlighter.java

```diff
@@ -72,6 +72,7 @@ public HighlightField highlight(FieldHighlightContext fieldContext) throws IOExc
         FetchSubPhase.HitContext hitContext = fieldContext.hitContext;
         MappedFieldType fieldType = fieldContext.fieldType;
         boolean forceSource = fieldContext.forceSource;
+        boolean fixBrokenAnalysis = fieldContext.context.mapperService().containsBrokenAnalysis(fieldContext.fieldName);
```

Comment:
this is introducing another usage of mapperService that we'd like to remove :( could we find a way around it?

Reply:
I added a delegator method, `FetchContext#containsBrokenAnalysis()`